indra/newview/llappviewerlinux.cpp

```diff
@@ -114,22 +156,75 @@ LLAppViewerLinux::~LLAppViewerLinux()
 {
 }
 
-bool LLAppViewerLinux::init()
+#if LL_SEND_CRASH_REPORTS
+std::string gCrashLogger;
+std::string gVersion;
+std::string gBugsplatDB;
+std::string gCrashBehavior;
+
+static bool dumpCallback(const google_breakpad::MinidumpDescriptor& descriptor, void* context, bool succeeded)
 {
-	// g_thread_init() must be called before *any* use of glib, *and*
-	// before any mutexes are held, *and* some of our third-party
-	// libraries likes to use glib functions; in short, do this here
-	// really early in app startup!
-	if (!g_thread_supported ()) g_thread_init (NULL);
+	if( fork() == 0 )
+		execl( gCrashLogger.c_str(), gCrashLogger.c_str(), descriptor.path(), gVersion.c_str(), gBugsplatDB.c_str(),  gCrashBehavior.c_str(), nullptr );
+    return succeeded;
+}
+
+void setupBreadpad()
```

Comment:
That's some ancient left overs. Later BugSplat tacked on top of it for crash reporting.

I am pretty sure this will not work right now (missing breakpad 3P to begin with).

Regarding crash reporting future (rip out or make it work) that's something LL will have to decide on.

If you think for not it's best to pull it out, I can do that (likely in another PR to not complicate this one even more)
I don't mind leaving it be either.

M personal suggestion would be to rip it out in another PR. The Linux user base always had been pretty small and the TCO of supporting a breakpad 3P + crash reporting + backend is probably higher than it is worth (at least for now)

Reply:
Honestly, I think it will be easier to maintain if we unify to BugSplat. BugSplat is also compatible with Linux.
BugSplat recommends combining it with Crashpad when running on Linux.

https://docs.bugsplat.com/introduction/getting-started/integrations/desktop/linux

Reply:
Bugsplat is compatible as in you have to roll parts your own. You will not get the nice bugsplat libraries like you get for Windows/OSX.

While it is true the recommendation being Crashpad (the successor of breakpad) breakpad is slightly easier to compile and there is existing packages in autobuild format (I have a semi recent one somewhere).

In the end it's just a choice and both will work, but [linux_crashlogger](https://github.com/secondlife/viewer/tree/main/indra/linux_crash_logger) will need a rewrite